cni/pkg/ambient/iptables.go

```diff
@@ -27,52 +29,62 @@ type iptablesRule struct {
 	RuleSpec []string
 }
 
+var kubeletChainsRegex = regexp.MustCompile(`(?m)^:(KUBE-IPTABLES-HINT|KUBE-KUBELET-CANARY)`)
+
+// hasKubeletChains checks if the output of an iptables*-save command
+// contains any of the rules set by kubelet.
+func hasKubeletChains(output []byte) bool {
+	return kubeletChainsRegex.Match(output)
+}
+
 // detectIptablesCommand will attempt to detect whether to use iptables-legacy, iptables or iptables-nft
 // based on output of iptables-nft or if the command exists.
 //
-// Logic is based on Kubernetes https://github.com/danwinship/kubernetes/blob/ca32fd23cca0797aa787fc5d883807d4eee6899f/build/debian-iptables/iptables-wrapper
-func (s *Server) detectIptablesCommand() string {
-	var err error
-	var numLegacyLines int
-	var numNftLines int
-	var output string
+// Logic is based on Kubernetes https://github.com/kubernetes-sigs/iptables-wrappers/blob/master/internal/iptables/detect.go
+func detectIptablesCommand() string {
+	// This method ignores all errors, this is on purpose. We execute all commands
+	// and try to detect patterns in a best effort basis. If something fails,
+	// continue with the next step. Worse case scenario if everything fails,
```

Comment:
Assuming this is matching K8S/kubelet behavior too - prefer `legacy` over `nft`.